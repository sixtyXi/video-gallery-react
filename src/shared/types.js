// @flow
import type { RecordFactory, RecordOf } from 'immutable';
import { Record } from 'immutable';

type MovieProps = {
  id: string | number,
  poster_path: string,
  title: string,
  genres: Array<string>,
  release_date: string,
  vote_average: number,
  tagline: string,
  runtime: number,
  overview: string
};

const defaultValuesMovie: MovieProps = {
  id: '',
  poster_path: '',
  title: '',
  genres: [],
  release_date: '',
  vote_average: 0,
  tagline: '',
  runtime: 0,
  overview: ''
};

export type Movie = RecordOf<MovieProps>;

export const makeMovie: RecordFactory<Movie> = Record(defaultValuesMovie);

export type Filter = {
  value: string,
  title: string
};

export type MovieFilters = {
  searchBy: string,
  sortBy: string,
  search: string
};

export type Location = {
  search: string,
  pathname: string
};
