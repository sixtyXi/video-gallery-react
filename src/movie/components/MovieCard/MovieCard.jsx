// @flow

import React from 'react';
import type { RecordOf } from 'immutable';
import type { Movie } from '../../../shared/types';

import styles from './MovieCard.scss';

export const MovieCard = ({ movie }: { movie: RecordOf<Movie> }) => {
  return (
    <article className={styles.wrapper}>
      <img
        className={styles.poster}
        src={movie.get('poster_path')}
        alt={`Poster of ${movie.get('title')}`}
      />
      <div className={styles.descWrapper}>
        <h1 className={styles.title} data-rate={movie.get('vote_average')}>
          {movie.get('title')}
        </h1>
        <p>{movie.get('tagline')}</p>
        <time className={styles.releaseDate}>
          {new Date(movie.get('release_date')).getFullYear()}
        </time>
        <p className={styles.runtime}>{`${movie.get('runtime')} min`}</p>
        <p>{movie.get('overview')}</p>
      </div>
    </article>
  );
};
