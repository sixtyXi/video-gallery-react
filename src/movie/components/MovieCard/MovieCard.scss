@import '../../../shared/_colors.scss';
@import '../../../shared/_typography.scss';

.wrapper {
  display: flex;
  width: 100%;
  margin: 20px 0;
  color: $text-light;
}

.poster {
  width: 200px;
  height: 280px;
}

.descWrapper {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  padding-left: 40px;
  font-size: $size-s;
  line-height: 1.5;

  & > * {
    width: 100%;
  }
}

.title {
  line-height: 1;
  color: $accent;
  position: relative;

  &:after {
    content: attr(data-rate);
    position: absolute;
    padding: 4px 2px;
    margin-left: 20px;
    font-size: $size-s;
    line-height: 1.16;
    color: $text-light;
    border: 1px solid;
    border-radius: 50%;
  }
}

.releaseDate,
.runtime {
  width: auto;
  margin-right: 20px;
  font-weight: bold;
}
