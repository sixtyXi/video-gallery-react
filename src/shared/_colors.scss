$accent: #fd4d61;
$secondary: #414141;
