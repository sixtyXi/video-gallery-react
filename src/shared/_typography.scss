$text-light: white;
$size-m: 14px;
$size-s: 12px;
$size-xs: 10px;
$size-l: 16px;
