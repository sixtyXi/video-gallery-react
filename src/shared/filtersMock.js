export const SORT_FILTERS = [
  {
    value: 'release_date',
    title: 'release date'
  },
  {
    value: 'vote_average',
    title: 'rating'
  }
];

export const SEARCH_FILTERS = [
  {
    value: 'title',
    title: 'title'
  },
  {
    value: 'genres',
    title: 'genre'
  }
];

export const DEFAULT_FILTERS = {
  searchBy: SEARCH_FILTERS[0].value,
  sortBy: SORT_FILTERS[0].value,
  search: ''
};
