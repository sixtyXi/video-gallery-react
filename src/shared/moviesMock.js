export const MOVIES = [
  {
    id: 680,
    title: 'Pulp Fiction',
    tagline: "Just because you are a character doesn't mean you have character.",
    vote_average: 8.3,
    vote_count: 10910,
    release_date: '1994-09-10',
    poster_path: 'https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg',
    overview:
      "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    budget: 8000000,
    revenue: 213928762,
    genres: ['Thriller', 'Crime'],
    runtime: 154
  },
  {
    id: 2,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 3,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 4,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 5,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 6,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 7,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  }
];
