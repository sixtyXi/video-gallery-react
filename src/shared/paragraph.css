.paragraph {
  height: 50px;
  display: inline-block;
  padding: 10px;
  line-height: 30px;
}
