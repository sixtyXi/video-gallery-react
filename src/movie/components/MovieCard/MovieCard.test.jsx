import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { MovieCard } from './MovieCard';
import { MOVIES } from '../../../shared/moviesMock';

describe('MovieCard', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<MovieCard movie={fromJS(MOVIES[0])} />);

    expect(wrapper).toMatchSnapshot();
  });
});
