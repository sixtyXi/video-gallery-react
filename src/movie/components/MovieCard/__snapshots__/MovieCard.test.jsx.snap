// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<article
  className="wrapper"
>
  <img
    alt="Poster of Four rooms"
    className="poster"
    src="https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg"
  />
  <div
    className="descWrapper"
  >
    <h1
      className="title"
      data-rate={3}
    >
      Four rooms
    </h1>
    <p>
      Oscar-winning Movies
    </p>
    <time
      className="releaseDate"
    >
      1995
    </time>
    <p
      className="runtime"
    >
      97 min
    </p>
    <p>
      Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.
    </p>
  </div>
</article>
`;
